packages/devtools_app/lib/src/history_manager.dart

```diff
@@ -0,0 +1,87 @@
+// Copyright 2021. The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+class HistoryManager<T> extends ChangeNotifier
```

Comment:
can you rename this to HistoryNotifier to make it clear that this is a notifier?

Reply:
actually on second thought, it seems strange that this class is a notifier at all? Why can't we just have this class be a normal dart class that has a value notifiers for the data other parts of the UI are subscribing to? then all the methods in here can update those notifiers accordingly